### ISSUE TITLE
Sidebar Broken on 0.2.0 Beta

### ISSUE THREAD

**Issue by AdmiralAsshat** [FreeTubeApp/FreeTube#65]
## Synopsis

The 0.2.0 Beta breaks the sidebar and all associated components within the sidebar.

## Environment:

My primary machine is a Dell XPS 13 9343 laptop running a flavor of Fedora with Cinnamon DE.

```
[13:09 guest ~]$ uname -a
Linux localhost 4.15.9-200.fc26.x86_64 #1 SMP Mon Mar 12 17:11:43 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
```
## Install:
```
[13:17 guest ~]$ rpm -qi FreeTube
Name        : FreeTube
Version     : 0.2.0
Release     : 1
Architecture: x86_64
Install Date: Sun 25 Mar 2018 01:06:15 PM EDT
Group       : Unspecified
Size        : 150951998
License     : GPL-3.0
Signature   : (none)
Source RPM  : FreeTube-0.2.0-1.src.rpm
Build Date  : Tue 20 Mar 2018 02:20:28 PM EDT
Build Host  : preston-gnome
Relocations : (not relocatable)
URL         : https://github.com/FreeTubeApp/FreeTube
Summary     : An Open Source YouTube app for privacy.
Description :
An Open Source YouTube app for privacy.
```

### II. Add some channels to subscribe to. 

Sidebar will work with first launch and session and all sidebar options will behave as expected.

### III. Close app and re-launch.

On subsequent re-launch, the sidebar will not respond to mouse-clicks. The middle of the screen will display the pulsating loading dot ad infinitum. 

![pulsating red dot](https://user-images.githubusercontent.com/5513487/37878147-40520ec2-3032-11e8-8c59-755a2174cadb.png)

## Remediation Attempts

 * My original machine updated to 0.2.0, so I tried uninstalling it and reinstalling from scratch. This did not help. My existing subscriptions immediately showed up, so it would seem that the app does not clean up all of its existing user data on uninstall. I can open a separate bug for this, if you wish.
 * Assuming that the pulsating dot was a load screen and that maybe the API key had been maxed out, I tried to assign my own key, but since I can't get to the Settings Menu, that did not work.
 * Did a fresh install on a VM of comparable distro. Sidebar worked and persisted through multiple sessions, so long as I did not add anything into my subscriptions. Once I added a channel and closed / reopened, the sidebar stopped working.

### Fedora XFCE
Below is from my Fedora desktop running Xfce. Again, it worked for several close/re-open sessions simply viewing individual videos, but broke once I added a subscription:

![broken in vm](https://user-images.githubusercontent.com/5513487/37878192-0e2a6da8-3033-11e8-9741-93a37629403f.png)

### Windows 7

I was able to duplicate the issue on a Windows 7 laptop as well, so this is not confined strictly to Linux builds.

![windows 7 screenshot](https://user-images.githubusercontent.com/5513487/37878202-4b20c9fa-3033-11e8-90c6-dfce2ee9e33e.PNG)

Please let me know if you need any other information. All of my dev tools are on Linux, so if you need me to run GDB or debug commands, I would be better-suited to doing so on the Linux machine.

**Comment by PrestonN** [comment]
Hey there.  Thanks for the very detailed bug report.  I was able to duplicate the issue and I will take a look at it.  It looks to be an issue when there are only a small amount of subscriptions (looks to be around 3 for me) so a temporary fix is to subscribe to a couple of more channels for now.

**Comment by AdmiralAsshat** [comment]
Well, I tried to add a few more, but it looks like the Search bar is unresponsive as well once borked.

I managed to work around it in the meantime by removing the subscriptions database:
`rm ~/.config/FreeTube/subscriptions.db`

That got it back to a fresh slate and I could re-add them from there.

**Comment by PrestonN** [comment]
I should have a fix for this coming out this weekend.  Let me know if it fixes the issue when it comes out.  There is more information at #70.

**Comment by AdmiralAsshat** [comment]
This appears to be fixed in version 0.2.1. Thanks!
